src/java.base/linux/classes/jdk/internal/platform/CgroupSubsystemFactory.java

```diff
@@ -128,6 +156,33 @@ static CgroupMetrics create() {
         return Optional.of(result);
     }
 
+    private static boolean isRelevantControllerMount(String line) {
+         Matcher lineMatcher = MOUNTINFO_PATTERN.matcher(line.trim());
+         boolean relevantControllerFound = false;
+         if (lineMatcher.matches()) {
+             String mountPoint = lineMatcher.group(1);
+             String fsType = lineMatcher.group(2);
+             if (fsType.equals("cgroup")) {
+                 String filename = Paths.get(mountPoint).getFileName().toString();
+                 for (String fn: filename.split(",")) {
+                     switch (fn) {
+                         case MEMORY_CTRL: // fall through
+                         case CPU_CTRL:
+                         case CPUSET_CTRL:
+                         case CPUACCT_CTRL:
+                         case BLKIO_CTRL:
+                             relevantControllerFound = true;
+                             break;
+                         default: break; // ignore not recognized controllers
+                     }
+                 }
+             } else if (fsType.equals("cgroup2")) {
+                 relevantControllerFound = true;
+             }
+         }
+         return relevantControllerFound;
```

Comment:
If I am reading the code right, then `relevantControllerFound` can be replaced with just `return true` or `return false`?

Reply:
I don't think so. It starts out as `false` and is being set to true on lines 180 and 174. So we could change it to `return false` here iff lines 174 and 180 would `return true`.

Reply:
Yes, that's what I meant: instead of dragging the boolean flag, do the early `return true`, or `return false` otherwise. Your choice if you want to make that change.

Reply:
OK, done. Thanks for the review!